packages/react-shortcuts/src/Shortcut/hooks.ts

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import {ShortcutContext} from '../ShortcutProvider';
+import Key, {HeldKey} from '../keys';
+
+export interface Subscription {
+  unsubscribe(): void;
+}
+
+export interface Options {
+  held?: HeldKey;
+  node?: HTMLElement | null;
+  ignoreInput?: boolean;
+  allowDefault?: boolean;
+}
+
+export default function useShortcut(
+  ordered: Key[],
+  onMatch: (matched: {ordered: Key[]; held?: HeldKey}) => void,
+  options: Options = {},
+) {
+  const context = React.useContext(ShortcutContext);
+  const subscription = React.useRef<Subscription | null>(null);
+
+  React.useEffect(
+    () => {
+      const {node} = options;
+      if (node != null) {
+        return;
+      }
+
+      if (context.shortcutManager == null) {
```

Comment:
So the context this gives is an object that has a `shortcutManager` property? Seems super weird...